platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/LdapLoginConfig.java

```diff
@@ -79,12 +111,128 @@
+                createGenericPoolConfig(id),
+                createGenericPoolAbandonConfig());
+
+      } catch (LdapException e) {
+        LOGGER.error("Error creating ldap connection factory", e);
+      }
+
+      // TODO Switch to DictionaryMap.java
+      Dictionary<String, String> serviceProps = new Hashtable<>();
+      serviceProps.put("id", id);
+
+      LOGGER.debug("Registering LdapConnectionPool");
+      connectionPoolServiceRegistration =
+          context.registerService(
+              GenericObjectPool.class.getName(), ldapConnectionPool, serviceProps);
       // create modules from the newly updated config
       Module ldapModule = createLdapModule(props);
       ldapService.update(ldapModule);
     }
   }
 
+  private AbandonedConfig createGenericPoolAbandonConfig() {
+    AbandonedConfig abandonedConfig = new AbandonedConfig();
+    abandonedConfig.setRemoveAbandonedOnBorrow(true);
+    abandonedConfig.setRemoveAbandonedTimeout(FIVE_MIN_S);
+    return abandonedConfig;
+  }
+
+  private GenericObjectPoolConfig createGenericPoolConfig(String id) {
+    GenericObjectPoolConfig config = new GenericObjectPoolConfig();
+    config.setJmxNameBase("org.apache.commons.pool2:type=LDAPConnectionPool,name=");
+    config.setJmxNamePrefix(id);
+    config.setTimeBetweenEvictionRunsMillis(FIVE_MIN_MS);
+    config.setTestWhileIdle(true);
+    config.setTestOnBorrow(true);
+    return config;
+  }
+
+  protected LDAPConnectionFactory createLdapConnectionFactory(String url, Boolean startTls)
+      throws LdapException {
+    boolean useSsl = url.startsWith("ldaps");
+    boolean useTls = !url.startsWith("ldaps") && startTls;
+
+    Options lo = Options.defaultOptions();
+
+    try {
+      if (useSsl || useTls) {
+        LOGGER.trace("Setting up secure LDAP connection.");
+        initializeSslContext();
+        lo.set(LDAPConnectionFactory.SSL_CONTEXT, getSslContext());
+      } else {
+        LOGGER.trace("Setting up insecure LDAP connection.");
+      }
+    } catch (GeneralSecurityException e) {
+      LOGGER.info("Error encountered while configuring SSL. Secure connection will fail.", e);
+    }
+
+    lo.set(LDAPConnectionFactory.HEARTBEAT_TIMEOUT, new Duration(30L, TimeUnit.SECONDS));
+    lo.set(LDAPConnectionFactory.HEARTBEAT_INTERVAL, new Duration(60L, TimeUnit.SECONDS));
+    lo.set(LDAPConnectionFactory.CONNECT_TIMEOUT, new Duration(30L, TimeUnit.SECONDS));
+
+    lo.set(LDAPConnectionFactory.SSL_USE_STARTTLS, useTls);
+    lo.set(
+        LDAPConnectionFactory.SSL_ENABLED_CIPHER_SUITES,
+        Arrays.asList(System.getProperty("https.cipherSuites", "").split(",")));
+    lo.set(
+        LDAPConnectionFactory.SSL_ENABLED_PROTOCOLS,
+        Arrays.asList(System.getProperty("https.protocols", "").split(",")));
+    lo.set(
+        LDAPConnectionFactory.TRANSPORT_PROVIDER_CLASS_LOADER,
+        SslLdapLoginModule.class.getClassLoader());
+
+    String host = url.substring(url.indexOf("://") + 3, url.lastIndexOf(':'));
```

Comment:
❓ why not use http://docs.oracle.com/javase/7/docs/api/java/net/URI.html for this rather than parsing it yourself?